### PR TITLE
document start doc only file factory

### PR DESCRIPTION
**Pull request by awalter-bnl**
updating the readme to include a description of how to create a factory that writes only the start doc.

This update will resolve issue #14

### CODE REVIEW

**Comment by awalter-bnl** [comment]
@danielballan this is ready for review

File: README.md [hunk 1]
```diff
@@ -7,3 +7,105 @@ This is a suitcase subpackage for reading a particular file format.
...
+    # ensure that any further documents from the RunRouter are processed.
+    return [serializer], []
+
+# generate the RunRouter from the factory and subscribe it too the RunEngine
+rr = event_model.RunRouter([factory])
+RE.subscribe(rr)
+```
+
+For a simple 'count' scan the file output will look like the following with
+each document type ('start', 'descriptor', 'event-page' or 'stop') found on a
```
Comment (danielballan) [inline README.md]:
```suggestion
each document type found on a
```
The comprehensive list would also include event, datum, datum_page, and resource (where applicable). Maybe better just not to enumerate them here.

File: README.md [hunk 2]
```diff
@@ -7,3 +7,105 @@ This is a suitcase subpackage for reading a particular file format.
 ```
 pip install suitcase-jsonl
 ```
+
+## Common use cases.
```
Comment (danielballan) [inline README.md]:
```suggestion
## Common use cases
```

**Comment by awalter-bnl** [comment]
comments by @danielballan have been fixed

File: README.md [hunk 3]
```diff
@@ -7,3 +7,104 @@ This is a suitcase subpackage for reading a particular file format.
 ```
 pip install suitcase-jsonl
 ```
+
+## Common use cases
+The common use case for this file format in streaming mode is to attach a
+RunRouter to the RunEngine (RE) using the following lines:
+
+```
```
Comment (mrakitin) [inline README.md]: This will add syntax highlighting:
```suggestion
```python
```

File: README.md [hunk 4]
```diff
@@ -7,3 +7,104 @@ This is a suitcase subpackage for reading a particular file format.
...
+# create a factory that will use the serializer to create on-the-fly files.
+def factory(name, start_doc):
+    # the following creates a serializer that will close the file when the stop
+    # document is sent to it.
+    serializer=suitcase.jsonl.Serializer(directory='...',
```
Comment (mrakitin) [inline README.md]:
```suggestion
    serializer = suitcase.jsonl.Serializer(directory='...',
```

File: README.md [hunk 5]
```diff
@@ -7,3 +7,104 @@ This is a suitcase subpackage for reading a particular file format.
...
+                                         file_prefix='...')
+    serializer(name, start_doc)
+    # ensure that any further documents from the RunRouter are processed.
+    return [serializer], []
+
+# generate the RunRouter from the factory and subscribe it too the RunEngine
```
Comment (mrakitin) [inline README.md]:
```suggestion
# generate the RunRouter from the factory and subscribe it to the RunEngine
```

File: README.md [hunk 6]
```diff
@@ -7,3 +7,104 @@ This is a suitcase subpackage for reading a particular file format.
...
+rr = event_model.RunRouter([factory])
+RE.subscribe(rr)
+```
+
+For a simple 'count' scan the file output will look like the following with
+each document type found on a new line in the file:
+
+```
```
Comment (mrakitin) [inline README.md]:
```suggestion
```python
```

File: README.md [hunk 7]
```diff
@@ -7,3 +7,104 @@ This is a suitcase subpackage for reading a particular file format.
...
+                "filled": {}, "data": {"det": [1.0]},
+                "timestamps": {"det": [1560865593.1378844]}}]
+["stop", {"run_start": "cd92fe6d-bcef-44cf-9624-205d7faa8164",
+          "time": 1560865593.1410978,
+          "uid": "7e43d05b-50b8-4b97-b4d0-41c7ffb778d1",
+          "exit_status": "success", "reason": "",
+          "num_events": {"primary": 1}}]
+```
+
+If for some reason only the start document is required then the following can
```
Comment (mrakitin) [inline README.md]:
```suggestion
If for some reason only the start document is required, then the following can
```

File: README.md [hunk 8]
```diff
@@ -7,3 +7,104 @@ This is a suitcase subpackage for reading a particular file format.
...
+be used to create the factory:
+
+```
```
Comment (mrakitin) [inline README.md]:
```suggestion
```python
```

**Comment by awalter-bnl** [comment]
Thanks for the review @mrakitin I have also commited your suggested changes

**Comment by danielballan** [comment]
Hm... Taking a step back, the line-breaks in JSONL are a critical part of the format. The examples shown add extra line-breaks for readability, but constitute invalid would-be examples of the output. I think we have to remove the line-breaks.

File: README.md [hunk 9]
```diff
@@ -7,3 +7,104 @@ This is a suitcase subpackage for reading a particular file format.
...
+The common use case for this file format in streaming mode is to attach a
+RunRouter to the RunEngine (RE) using the following lines:
+
+```python
+# create a factory that will use the serializer to create on-the-fly files.
+def factory(name, start_doc):
+    # the following creates a serializer that will close the file when the stop
+    # document is sent to it.
+    serializer = suitcase.jsonl.Serializer(directory='...',
+                                         file_prefix='...')
```
Comment (mrakitin) [inline README.md]: Thanks for applying my suggestions, @awalter-bnl! I missed one more line after adding a couple of blank spaces:
```suggestion
                                           file_prefix='...')
```

**Comment by awalter-bnl** [comment]
>Hm... Taking a step back, the line-breaks in JSONL are a critical part of the format. The examples shown add extra line-breaks for readability, but constitute invalid would-be examples of the output. I think we have to remove the line-breaks.

@danielballan, I can do that but then the examples are next to useless as they are not readable. So I would rather not have them than include them without the readability line breaks

**Comment by danielballan** [comment]
OK. I am arguing than wrong examples are worse than no examples, and you are arguing that examples that don’t fit on a readable line are worse than no examples. Should we use `...` to abbreviate the examples? Link to an example output file rather than including it directly in the text? Omit examples entirely?

**Comment by awalter-bnl** [comment]
Due to a lack of consensus on how to implement this I think it is unlikely to get through so am closing it out.

**Pull request closed without merging**